runelite-client/src/main/java/net/runelite/client/plugins/config/ProfilePanel.java

```diff
@@ -666,6 +724,74 @@ private void switchToProfile(long id)
 		executor.submit(() -> configManager.switchProfile(profile));
 	}
 
+	private void unsetRsProfileDefaultProfile()
+	{
+		setRsProfileDefaultProfile(-1);
+	}
+
+	private void setRsProfileDefaultProfile(long id)
+	{
+		boolean switchProfile = false;
+		try (ProfileManager.Lock lock = profileManager.lock())
```

Comment:
Profile lock should only happen on the executor thread. This should probably live in ProfileManager, not in here

Reply:
`(create|delete|rename|switchTo)Profile` and `handleDrag` don't create a lock on the executor thread, though I do see `importProfile` and `cloneProfile` do, would you want all of these to run on the executor thread also?

I'm not really sure how this functionality could or should move to ProfileManager... Profile changes trigger no events, so even if all the logic around setting and unsetting a profile as default for some rsprofile was moved into ProfileManager, there wouldn't be a good way to react to the changes it would enact. As-is, `setRsProfileDefaultProfile()` effectively does the same work as `switchToProfile()`, albeit on a different ConfigProfile field.

Reply:
I think I am remembering something that I wanted to change but didn't, not sure. This is fine